src/ledger/LedgerTxnDataSQL.cpp

```diff
@@ -410,17 +351,13 @@ LedgerTxnRoot::Impl::encodeDataNamesBase64()
         session << "ALTER TABLE accountdata ALTER COLUMN dataname "
                    "SET DATA TYPE VARCHAR(88)";
     }
-
-    size_t numUpdated = 0;
-    for (auto const& le : dataToEncode)
+    if (!dataToEncode.empty())
     {
-        insertOrUpdateData(le, true);
-
-        if ((++numUpdated & 0xfff) == 0xfff ||
-            (numUpdated == dataToEncode.size()))
-        {
-            CLOG(INFO, "Ledger") << "Wrote " << numUpdated << " data entries";
-        }
+        BulkUpsertDataOperation op(mDatabase, dataToEncode.begin(),
```

Comment:
Looks like when I commented about this earlier, I was imagining it working differently from you. I was thinking of doing batches of size `LEDGER_ENTRY_BATCH_COMMIT_SIZE`, which would have been easier with iterators. I think I was making it a little more complicated than we needed, because the number of data entries is very small so I think we can do them in a single operation as you do here. No need to do it with iterators, might as well just pass `dataToEncode` by reference instead. Sorry for the confusion.

Reply:
Oh, ok! fixed to use plain vectors and a single-entry accumulate helper.